src/plugins/kibana_react/public/code_editor/code_editor.tsx

```diff
@@ -102,116 +110,244 @@ export interface Props {
   fullWidth?: boolean;
 }
 
-export class CodeEditor extends React.Component<Props, {}> {
-  _editor: monaco.editor.IStandaloneCodeEditor | null = null;
-
-  _editorWillMount = (__monaco: unknown) => {
-    if (__monaco !== monaco) {
-      throw new Error('react-monaco-editor is using a different version of monaco');
-    }
-
-    if (this.props.overrideEditorWillMount) {
-      this.props.overrideEditorWillMount();
-      return;
+export const CodeEditor: React.FC<Props> = ({
+  languageId,
+  value,
+  onChange,
+  width,
+  height,
+  options,
+  overrideEditorWillMount,
+  editorDidMount,
+  editorWillMount,
+  useDarkTheme,
+  transparentBackground,
+  suggestionProvider,
+  signatureProvider,
+  hoverProvider,
+  languageConfiguration,
+}) => {
+  // We need to be able to mock the MonacoEditor in our test in order to not test implementation
+  // detail and not have to call methods on the <CodeEditor /> component instance.
+  const MonacoEditor: typeof ReactMonacoEditor = useMemo(() => {
+    const isMockedComponent =
+      typeof ReactMonacoEditor === 'function' && ReactMonacoEditor.name === 'JestMockEditor';
+    return isMockedComponent ? (ReactMonacoEditor as any)() : ReactMonacoEditor;
+  }, []);
+
+  const _editor = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
+  const isSuggestionMenuOpen = useRef(false);
+  const editorHint = useRef<HTMLDivElement>(null);
+
+  const [isHintActive, setIsHintActive] = useState(true);
+
+  /* eslint-disable @typescript-eslint/naming-convention */
+  const promptClasses = classNames('kibanaCodeEditor__keyboardHint', {
+    'kibanaCodeEditor__keyboardHint--isInactive': !isHintActive,
+  });
+  /* eslint-enable  @typescript-eslint/naming-convention */
+
+  const _updateDimensions = useCallback(() => {
+    if (_editor.current) {
+      _editor.current.layout();
     }
+  }, []);
+
+  const startEditing = useCallback(() => {
+    setIsHintActive(false);
+    _editor.current?.focus();
+  }, []);
+
+  const stopEditing = useCallback(() => {
+    setIsHintActive(true);
+  }, []);
+
+  const onKeyDownHint = useCallback(
+    (ev) => {
+      if (ev.keyCode === keyCodes.ENTER) {
```

Comment:
I think we're trying to use `ev.key` nowadays more. Also in this case you can import `keys` from EUI, which contains a map of well readable names (that you created above).

Reply:
I am not seeing the `key` prop on the event. The `keys` from EUI gives me back "Escape" (string) for `keys.ESCAPE` and not the key code. Is that expected?

Reply:
As solved offline: Monaco's keycodes have nothing to do with JavaScript keycodes, and thus we should refer to their map: https://microsoft.github.io/monaco-editor/api/enums/monaco.keycode.html

Reply:
@sebelga Sorry I got confused and didn't read your answer properly. I I think on this event there should be a `key` property? This should be a regular `React.KeyboardEvent`. And yes, `key` would be a string representation of the key and thus match `keys.ESCAPE` from EUI.

Let's please also make sure everything here is typed properly, by actually giving this `ev` the `React.KeyboardEvent` type, so TS would complain about mismatches.